### ISSUE TITLE
Custom conversion exceptions wrapping

### ISSUE THREAD

**Issue by dscheg** [paiden/Nett#29]
It's useful to show config line and column numbers in case of failed custom conversion.
```
TomlConfig.Create(cfg =>
    cfg.ConfigureType<Color>(type =>
        type.WithConversionFor<TomlString>(conv =>
            conv.FromToml(item => ColorHelper.Parse(item.Value)))));
```

Something like that (conversion exception catched and wrapped):
```
System.Exception: Line 41, Column 15: Conversion of value '#1234' to 'Color' type failed.
```

But now user can only get:
```
System.FormatException: Failed to parse Color '#1234'
   at X.ColorHelper.Parse(String color)
   at Nett.TomlConverterBase`2.Convert(ITomlRoot root, Object o, Type targetType)
   at Nett.TomlValue`1.Get(Type t)
   at Nett.TomlTable.Get(Type t)
   at Nett.TomlTable.Get(Type t)
   at Nett.TomlObject.Get[T]()
```

**Comment by paiden** [comment]
I agree that the conversion error messages should be improved. I will try to find ways to do so in the future.

But catching and wrapping the exception in the format you mentioned will not be possible, because there is no parse issue. When the conversion takes place the parser has already finished processing the file. In fact when the conversion from a TomlTable to a CLR object is done, Nett doesn't even know that the Table was created from a file.

E.g. the following example will also throw a conversion exception. This sample does no parsing at all.
```
public class Item
{
    public int Value { get; set; }

    public static Item Parse(string s)
    {
        throw new Exception("Simulate conversion error");
    }
}

public class Root
{
    public Item Item { get; set; }
}

[Fact]
public void Foo()
{
    var cfg = TomlConfig.Create(c =>
        c.ConfigureType<Item>(type =>
            type.WithConversionFor<TomlString>(conv =>
                conv.FromToml(s => Item.Parse(s.Value)))));
    TomlTable tbl = Toml.Create(cfg);
    tbl.AddValue("Item", "Item Parse Value");

    var clrObject = tbl.Get<Root>();
}
```

**Comment by paiden** [comment]
Fixed in 0.12.0
